app/ts/background/windows/confirmTransaction.ts

```diff
@@ -166,106 +168,71 @@ export async function openConfirmTransactionDialog(
-		if (activeAddress === undefined) return ERROR_INTERCEPTOR_NO_ACTIVE_ADDRESS
-
-		const addedPendingTransaction = await pendingConfirmationSemaphore.execute(async () => {
-			if (!justAddToPending) {
-				const pendingTransactions = await getPendingTransactions()
-				if (pendingTransactions.length !== 0) {
-					const pendingTransaction = pendingTransactions[0]
-					if (pendingTransaction === undefined) throw new Error('pending transaction was undefined')
-					if (await getPopupOrTabOnlyById(pendingTransaction.popupOrTabId) !== undefined) {
-						justAddToPending = true
-					} else {
-						await clearPendingTransactions()
-					}
-				}
-			}
-
-			const openDialog = async () => {
-				addWindowTabListeners(onCloseWindow, onCloseTab)
-				return await openPopupOrTab({ url: getHtmlFile('confirmTransaction'), type: 'popup', height: 800, width: 600 })
-			}
-			if (!justAddToPending || openedDialog === undefined) openedDialog = await openDialog()
-			else { await tryFocusingTabOrWindow(openedDialog) }
-
-			if (openedDialog === undefined) return false
-			const pendingTransaction = {
-				popupOrTabId: openedDialog,
-				originalRequestParameters: transactionParams,
-				uniqueRequestIdentifier: request.uniqueRequestIdentifier,
-				simulationMode,
-				activeAddress,
-				created,
-				status: 'Crafting Transaction' as const,
-				transactionIdentifier,
-				website,
+	const created = new Date()
+	const transactionToSimulatePromise = transactionParams.method === 'eth_sendTransaction' ? formEthSendTransaction(ethereumClientService, activeAddress, simulationMode, website, transactionParams, created, transactionIdentifier) : formSendRawTransaction(ethereumClientService, transactionParams, website, created, transactionIdentifier)
+	if (activeAddress === undefined) return { type: 'result' as const, ...ERROR_INTERCEPTOR_NO_ACTIVE_ADDRESS }
+
+	const addedPendingTransaction = await pendingConfirmationSemaphore.execute(async () => {
+		const getPendingTransactionWindow = async () => {
+			const pendingTransactions = await getPendingTransactions()
+			const [firstPendingTransaction] = pendingTransactions
+			if (firstPendingTransaction !== undefined) {
+				const alreadyOpenWindow = await getPopupOrTabOnlyById(firstPendingTransaction.popupOrTabId)
+				if (alreadyOpenWindow) return alreadyOpenWindow
+				await resolvePendingTransactionsAsNoResponse(pendingTransactions, simulator, websiteTabConnections)
 			}
-			await appendPendingTransaction(pendingTransaction)
-			if (justAddToPending) await sendPopupMessageToOpenWindows({ method: 'popup_confirm_transaction_dialog_pending_changed', data: { pendingTransactions: await getPendingTransactions() } })
-				
+			return await openPopupOrTab({ url: getHtmlFile('confirmTransaction'), type: 'popup', height: 800, width: 600 })
+		}
+		const openedDialog = await getPendingTransactionWindow()
+		if (openedDialog === undefined) return false //todo add better error here, somehow we failed to create window
```

Comment:
maybe switch the comment to github issue if this is already tracked

Reply:
its already handled in the most recent pr